src/basic/time-util.h

```diff
@@ -48,6 +48,7 @@ typedef struct triple_timestamp {
         usec_t boottime;
 } triple_timestamp;
 
+#define SEC_INFINITY  ((uint32_t) -1)
```

Comment:
i don't think we should this macro. The type is too unspecific I fear. People store seconds in any kind of types, not just uint32_t. In fact the ndisc lifetime is actually a uint16_t, hence there's already confusion about the type within this very patch...

Moreover I am pretty sure we should stick to usec_t times wherever appropriate, in particular in generic code, hence the Route's lifetime field should be changed to usec_t even if it is initialized from the ndisc lifetime value (that is if we need the field at all. But i have the suspicion we should drop it altogether since it is redundant)